engine/table/src/main/java/io/deephaven/engine/table/impl/select/WhereFilterPatternImpl.java

```diff
@@ -0,0 +1,181 @@
+/**
+ * Copyright (c) 2016-2022 Deephaven Data Labs and Patent Pending
+ */
+package io.deephaven.engine.table.impl.select;
+
+import io.deephaven.api.ColumnName;
+import io.deephaven.api.filter.FilterPattern;
+import io.deephaven.api.filter.FilterPattern.Mode;
+import io.deephaven.chunk.LongChunk;
+import io.deephaven.chunk.ObjectChunk;
+import io.deephaven.chunk.WritableLongChunk;
+import io.deephaven.chunk.attributes.Values;
+import io.deephaven.engine.rowset.RowSet;
+import io.deephaven.engine.rowset.WritableRowSet;
+import io.deephaven.engine.rowset.chunkattributes.OrderedRowKeys;
+import io.deephaven.engine.table.ColumnDefinition;
+import io.deephaven.engine.table.ColumnSource;
+import io.deephaven.engine.table.Table;
+import io.deephaven.engine.table.TableDefinition;
+import io.deephaven.engine.table.impl.chunkfilter.ChunkFilter;
+import io.deephaven.engine.table.impl.chunkfilter.ChunkFilter.ObjectChunkFilter;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+import java.util.function.Predicate;
+
+final class WhereFilterPatternImpl extends WhereFilterImpl {
+
+    private static final long serialVersionUID = 1L;
+
+    public static WhereFilterPatternImpl of(FilterPattern pattern, boolean inverted) {
```

Comment:
We discussed that all of this inversion could be expressed as a filter that wraps the internal filter, and returns `selection.minus(wrapped.filter(...))`

Reply:
Done